### ISSUE TITLE
Pro - Progress bar values are about an order of magnitude off

### ISSUE THREAD

**Issue by elliotmatson** [ayushsharma82/ESP-DASH#244]
**Describe the bug**
Progress bar is showing something close to the update value / 10

**To Reproduce**
```
Card influxBufferSizeCard(&dashboard, PROGRESS_CARD, "Influx Buffer", " kB", 0, 1024);
...
influxBufferSizeCard.update(20, " kB / 1024 kB");
dashboard.refreshCard(&influxBufferSizeCard);
```
Results in:
![image](https://github.com/user-attachments/assets/53985f82-b4d4-449b-ba55-8f8e0d58519d)

Have tested with ints and floats, and finally with a constant. It doesn't seem to be exactly inputVal/10, but it's typically pretty close.
Logging the value of inputVal immediately before calling update() verifies that the expected value is correct.

**Expected behavior**
Should show 20

**Versions**
ESP-DASH Pro @ 4.6.0
bblanchon/ArduinoJson @ ^7.1.0
mathieucarbou/AsyncTCP @ 3.2.12
mathieucarbou/ESPAsyncWebServer @ 3.2

**Comment by mathieucarbou** [comment]
Are you sure to use int everywhere  and not floats ?

Sliders support both int and floats:

`Card _output1DimmerSlider = Card(&dashboard, SLIDER_CARD, YASOLR_LBL_050, "%", 0.0f, 100.0f, 0.01f);`

if you construct the card with integer, it will have internally: `_value_type = Card::INTEGER;`

with floats, it will have internally: `_value_type = Card::FLOAT;`

This distinction is important for ESP-DASH to know how to output the numbers, and which callback to call between:

```
    void attachCallback(std::function<void(int)> cb);
    void attachCallbackF(std::function<void(float)> cb);
```

Rule: always use the same types: int, or float, but not a mix of them otherwise it will change the card type at runtime.

Note: this might not be related to your issue though, but just something to check.

**Comment by elliotmatson** [comment]
Still seeing this issue

**Comment by elliotmatson** [comment]
Bump

**Comment by mathieucarbou** [comment]
> I definitely used all ints in my original code, but when testing a float in update() I didn't change the min and max to floats as well

yes, it is either everything floats (update ,callback AND definition), or everything in ints.

I do not have this behaviour in chrome and safari... really weird...

**Comment by elliotmatson** [comment]
That's interesting, if it isn't reproducible for anyone else it must be with my build environment or browser, I'll see if I can run some more tests

**Comment by ayushsharma82** [comment]
Ah, now I got to understand your issue. @elliotmatson  Please send a message over support@softt.io to create a ticket, I'll reply back to you with a fix there. I believe we can fix it today itself.

**Comment by elliotmatson** [comment]
Just sent, thanks so much!

**Comment by ayushsharma82** [comment]
@mathieucarbou  We were so focused on blaming the int and floats haha but the issue was that UI calculates the % of raw value in the specified range. So what @elliotmatson thought was off-value was just the percentage in range of 0-100. 

As mentioned on mail, the best fix is to keep the percentage but update the symbol string as well to update raw value. For example:

```c
String symbol_str = "% (" + String(value) + "KB / 1024KB)";
progress_card.update(value, symbol_str.c_str());
```
